src/intersection.js

```diff
@@ -23,5 +23,7 @@ var uniq = require('./uniq');
  *      R.intersection([1,2,3,4], [7,6,5,4,3]); //=> [4, 3]
  */
 module.exports = _curry2(function intersection(list1, list2) {
-  return uniq(_filter(flip(_contains)(list1), list2));
+  var lookupList = list1.length > list2.length ? list1 : list2;
+  var filteredList = lookupList == list1 ? list2 : list1;
```

Comment:
I'd prefer to branch once rather than twice:

```javascript
if (list1.length > list2.length) {
  // ...
} else {
  // ...
}
return ...;
```